hathor/transaction/base_transaction.py

```diff
@@ -820,6 +820,14 @@ def serialize_output(tx: BaseTransaction, tx_out: TxOutput) -> dict[str, Any]:
             'parents': [],
         }
 
+        if not self.is_block:
```

Comment:
If you need this information for nano, why don't you add it to the nano class `to_json_extended`?

Even if you prefer to add to all transactions classes, you could add it do the `Transaction` class, instead of doind this `if` here.

Reply:
Oh, I did it in the BaseTransaction class because Transaction doesn't have `to_json_extended` and in all the resources involving transaction we call `to_json_extended`.

In addition I couldn't find the nano class. I believe it is in the other repository.

I agree we can narrow down the scope of `first_block` and remove any block from the equation. However, either we add this property only for nano contract transaction, or to all transactions except blocks. What is best?

On the one side we have a better coherence when `first_block` appears only for nano contract transactions. On the other side, we have a better consistency when `first_block` appears for all transaction except block.

I prefer to keep consistency because it give us a degree of freedom for when the system needs to evolve, for instance a new transaction version that also requires a block confirmation.

Reply:
In the following commit https://github.com/HathorNetwork/hathor-core/pull/1003/commits/37a9515a0e28ef47515e86c1d76ab9552478af0a I removed `first_block` from `to_json_extended` of base transaction and I added `to_json_extended` method to transaction class.